### DISCUSSION TITLE
Entering inventory of items w/o prior qty

### DISCUSSION THREAD

**Discussion by kaare** [ledgersmb/LedgerSMB#5870] [Design]
1.5.6

When entering the report (Goods and services, Enter Inventory), it bombed out with a type mismatch on the expected value. Turns out we had stock on items with no prior onhand, so it was undefined.  Changing line 136 in lib/LedgerSMB/Inventory/Adjust.pm to
           expected => $hashref->{"onhand_$ln"} // 0,
solves the problem and I suggest that to be added to the next release. You can always discuss if it's an error to update an item w/o prior stock information, but failing with a Moose stacktrace isn't the best option, I think.

**Comment by ehuelsmann** [comment]
There's one problem with this change, I think, but maybe @einhverfr can comment on that: if there never was a prior invoice for the parts you're stocking, we're currently stocking them at a price of zero, which means your inventory balance and COGS postings for the sale of these parts are off until you sold them.

This is basically a question as to how to seed a company database with stock without entering all the invoices related to this stock (e.g. when setting up a company migrating from another accounting system).
